### PR TITLE
[CLI-312] Inconsistent behaviour in key/value pairs (Java property style)

### PR DESCRIPTION
**Pull request by Claudenw**
Fix for CLI-312 https://issues.apache.org/jira/browse/CLI-312

Solves issue where java property type option has a single value separated from the -D flag e.g "-D y" as was illustrated in pull request #189

closes #189

### CODE REVIEW

File: src/main/java/org/apache/commons/cli/DefaultParser.java [hunk 1]
```diff
@@ -228,6 +228,9 @@ private DefaultParser(final boolean allowPartialMatching,
      */
     private void checkRequiredArgs() throws ParseException {
         if (currentOption != null && currentOption.requiresArg()) {
+            if (isJavaProperty(currentOption.getKey()) && currentOption.getValues().length == 1) {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/cli/DefaultParser.java:231]: Hi @Claudenw 
I think you are missing a test case to account for `Option#getValues()` returning `null`.

**Review by garydgregory — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/commons/cli/DefaultParser.java (hunk 1 above)
Comment (Claudenw) [reply src/main/java/org/apache/commons/cli/DefaultParser.java]: Added test, and modified code to handle the case.  Good catch.

File: src/main/java/org/apache/commons/cli/DefaultParser.java [hunk 2]
```diff
@@ -228,6 +228,9 @@ private DefaultParser(final boolean allowPartialMatching,
      */
     private void checkRequiredArgs() throws ParseException {
         if (currentOption != null && currentOption.requiresArg()) {
+            if (isJavaProperty(currentOption.getKey()) && currentOption.numberOfValues() == 1) {
```
[the remark this answers is not in the document]
Comment (Claudenw) [reply src/main/java/org/apache/commons/cli/DefaultParser.java]: I missed `getValuesList()` when I was implementing.  I saw that `getValues()` would return `null` and that threw me off.  Good catch.

File: src/test/java/org/apache/commons/cli/bug/BugCLI312Test.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                } else {
+                    fail("Didn't expect " + defineOptionsFound + " occurrences of -D");
+                }
+            }
+        }
+        assertEquals("other", cl.getArgList().get(0));
+    }
+
+    @Test
+    public void testNoOptionValues() throws ParseException {
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/cli/bug/BugCLI312Test.java:97]: `ParseException` is not thrown by this method, please remove it from the method signature.

**Review by garydgregory — CHANGES_REQUESTED** [review]
@Claudenw 
TY for your update.
One more comment...

File: src/test/java/org/apache/commons/cli/bug/BugCLI312Test.java (hunk 3 above)
Comment (Claudenw) [reply src/test/java/org/apache/commons/cli/bug/BugCLI312Test.java]: done
